Modified `MarginalDistribution`

akash9712 opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234". See
https://github.com/blog/1506-closing-issues-via-pull-requests .-->


#### Brief description of what is fixed or changed
Modified `MarginalDistribution` so that it can be initialised without
the the use of `RandomSymbols`, using a joint distribution and indices.
#### Other comments


#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. If there is no release notes entry for this PR,
write "NO ENTRY". The bot will check your release notes automatically to see
if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

Upabjojr commented on sympy/core/tests/test_args.py:
```diff
@@ -1054,10 +1054,10 @@ def test_sympy__stats__joint_rv__JointDistributionHandmade():
-    r = RandomSymbol(S('r'))
-    assert _test_args(MarginalDistribution(r, (r,)))
+    N1 = Normal('N1', [1, 0], [[1, 0], [0, 1]])
+    assert _test_args(MarginalDistribution(N1, (N1[1],)))
```
I would prefer to construct `MarginalDistribution` without occurrences of `RandomSymbol` in it.
Like:
```python
dist = N1.pspace.distribution
_test_args(MarginalDistribution(dist, (1,)))
```
Distribution objects should not contain `RandomSymbol` or `PSpace` objects, only the parameters defining the distribution.

Upabjojr commented on sympy/stats/joint_rv_types.py:
```diff
@@ -254,3 +253,50 @@ def NormalGamma(syms, mu, lamda, alpha, beta):
     """
     return multivariate_rv(NormalGammaDistribution, syms, mu, lamda, alpha, beta)
+
+#-----------------------STOOCHASTIC PROCESSES--------------------------------
```
What about a new file?

Upabjojr commented on sympy/stats/joint_rv.py:
```diff
@@ -330,3 +351,12 @@ def marginalise_out(self, expr, rv):
     def __call__(self, *args):
         return self.pdf(*args)
+
+class StochasticProcess(JointDistribution):
```
I don't think this is a joint distribution in the end. I should behave like a joint distribution in some cases.

jksuom replied:
I think it should rather be like collection of joint distributions. It should have a method creating a joint distribution for each finite set of indices.

Upabjojr commented on sympy/stats/joint_rv.py:
```diff
@@ -330,3 +351,12 @@ def marginalise_out(self, expr, rv):
+    """
+
+    @property
+    def component_count(self):
```
We may have countable and uncountable stochastic processes. We need to consider that.

akash9712 committed:
Modified `MarginalDistribution`

Modified `MarginalDistribution` so that it can be initialised without
the the use of `RandomSymbols`, using a joint distribution and indices.

akash9712 committed:
Merge branch 'master' into stochastic

czgdp1807 commented:
Since, `stats` module plans to implement this concept in another way, I am closing this PR for now. However, feel free to reuse it in your GSoC project. Thanks for the contributions.
